src/storage/secondary/rowset/rowset_iterator.rs

```diff
@@ -84,68 +81,51 @@ impl RowSetIterator {
             };
         }
 
-        let filter_expr = if let Some(expr) = expr {
-            let filter_column = expr.get_filter_column(column_refs.len());
-            // assert filter column is not all false
-            assert!(
-                filter_column.any(),
-                "There should be at least 1 filter column"
-            );
-            Some((expr, filter_column))
-        } else {
-            None
-        };
-
         Ok(Self {
             column_refs,
             dvs,
             column_iterators,
-            filter_expr,
-            start_keys: start_keys.to_vec(),
-            end_keys: end_keys.to_vec(),
-            meet_end_key_before: false,
-            meet_start_key_before: false,
+            filter,
+            end: false,
         })
     }
 
-    pub async fn next_batch_inner(
+    pub async fn next_batch(
         &mut self,
         expected_size: Option<usize>,
-    ) -> StorageResult<(bool, Option<StorageChunk>)> {
-        // We have met end key in pre `StorageChunk`
-        // so we can finish cur scan.
-        if self.meet_end_key_before {
-            return Ok((true, None));
+    ) -> StorageResult<Option<StorageChunk>> {
+        if self.end {
+            return Ok(None);
         }
-        let filter_context = self.filter_expr.as_ref();
         // It's guaranteed that `expected_size` <= the number of items left
         // in the current block, if provided
         let mut fetch_size = {
             // We find the minimum fetch hints from the column iterators first
-            let mut min = None;
+            let mut min: Option<usize> = None;
             let mut is_finished = true;
             for it in &self.column_iterators {
                 let (hint, finished) = it.fetch_hint();
                 if !finished {
-                    is_finished = false
+                    is_finished = false;
                 }
                 if hint != 0 {
-                    if min.is_none() {
-                        min = Some(hint);
+                    if let Some(v) = min {
+                        min = Some(v.min(hint));
                     } else {
-                        min = Some(min.unwrap().min(hint));
+                        min = Some(hint);
                     }
                 }
             }
 
-            if min.is_some() {
-                min.unwrap().min(ROWSET_MAX_OUTPUT)
+            if let Some(min) = min {
+                min.min(ROWSET_MAX_OUTPUT)
```

Comment:
should this be `if let Some(ref mut min) = min` and `*min = min.min(xxx)`?

Reply:
The whole if expression will be assigned to `fetch_size`. I think it is fine.